snuba/datasets/configuration/storage_builder.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+from typing import Any
+
+from jsonschema import validate
+from yaml import safe_load
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.datasets.configuration.json_schema import (
+    V1_READABLE_STORAGE_SCHEMA,
+    V1_WRITABLE_STORAGE_SCHEMA,
+)
+from snuba.datasets.configuration.utils import (
+    CONF_TO_PREFILTER,
+    CONF_TO_PROCESSOR,
+    generate_policy_creator,
+    get_query_processors,
+    parse_columns,
+)
+from snuba.datasets.schemas.tables import TableSchema, WritableTableSchema
+from snuba.datasets.storage import ReadableTableStorage, WritableTableStorage
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.table_storage import (
+    KafkaStreamLoader,
+    build_kafka_stream_loader_from_settings,
+)
+from snuba.subscriptions.utils import SchedulingWatermarkMode
+from snuba.utils.streams.topics import Topic
+
+KIND = "kind"
+WRITABLE_STORAGE = "writable_storage"
+STORAGE = "storage"
+STORAGE_KEY = "storage_key"
+SET_KEY = "set_key"
+SCHEMA = "schema"
+STREAM_LOADER = "stream_loader"
+PRE_FILTER = "pre_filter"
+QUERY_PROCESSORS = "query_processors"
+SUBCRIPTION_SCHEDULER_MODE = "subscription_scheduler_mode"
+DLQ_POLICY = "dlq_policy"
+
+# TODO: The config files should be discovered automatically
+CONFIG_FILES_PATH = "./snuba/datasets/configuration/generic_metrics/storages"
+CONFIG_FILES = {
+    StorageKey.GENERIC_METRICS_DISTRIBUTIONS: f"{CONFIG_FILES_PATH}/distributions.yaml",
+    StorageKey.GENERIC_METRICS_DISTRIBUTIONS_RAW: f"{CONFIG_FILES_PATH}/distributions_bucket.yaml",
+    StorageKey.GENERIC_METRICS_SETS_RAW: f"{CONFIG_FILES_PATH}/sets_bucket.yaml",
+    StorageKey.GENERIC_METRICS_SETS: f"{CONFIG_FILES_PATH}/sets.yaml",
+}
+
+STORAGE_VALIDATION_SCHEMAS = {
+    "readonly_storage": V1_READABLE_STORAGE_SCHEMA,
+    "writable_storage": V1_WRITABLE_STORAGE_SCHEMA,
+}
+
+
+def build_readonly_storage(storage_key: StorageKey) -> ReadableTableStorage:
+    config = __load_storage_config(storage_key)
+    storage_kwargs = __build_readonly_storage_kwargs(config)
+    return ReadableTableStorage(**storage_kwargs)
+
+
+def build_writable_storage(storage_key: StorageKey) -> WritableTableStorage:
```

Comment:
This doesn't seem like it should be based on `StorageKey` to me. Why not just take a `str` here? (same with `build_readonly_storage`). And `__load_storage_config` definitely seems like it should take a filename, not a storage key